clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -885,6 +885,18 @@ public static void closeAll(Closeable... closeables) throws IOException {
             throw exception;
     }
 
+    /**
+     * An {@link AutoCloseable} interface without a throws clause in the signature
+     *
+     * This is used with lambda expressions in try-with-resources clauses
+     * to avoid casting un-checked exceptions to checked exceptions unnecessarily.
+     */
+    @FunctionalInterface
+    public interface QuietClosable extends AutoCloseable {
```

Comment:
```suggestion
    public interface QuietClosable extends AutoCloseable {
```
I don't think the name is very accurate. The interface does not prevent implementation of close from throwing (as opposed to the methods below) and its demonstrated use does not either. (there's also a typo, if typos in non-words matter). 

How about `UncheckedCloseable`?